src/graph/heterograph_capi.cc

```diff
@@ -452,6 +452,31 @@ DGL_REGISTER_GLOBAL("heterograph_index._CAPI_DGLHeteroDisjointPartitionBySizes")
     *rv = ret_list;
 });
 
+DGL_REGISTER_GLOBAL("heterograph_index._CAPI_DGLHeteroGetSparseFormat")
+.set_body([] (DGLArgs args, DGLRetValue* rv) {
+    HeteroGraphRef hg = args[0];
+    dgl_type_t etype = args[1];
+    CHECK_LE(etype, hg->NumEdgeTypes()) << "invalid edge type " << etype;
+    auto bg = std::dynamic_pointer_cast<UnitGraph>(hg.sptr());
+    if (bg != nullptr)  // hg itself is a unit graph.
+      *rv = hg->GetSparseFormat();
+    else
+      *rv = hg->GetRelationGraph(etype)->GetSparseFormat();
```

Comment:
Will we still get a UnitGraph from Python? I thought I've patched everywhere ...